src/Bicep.Core/Syntax/ObjectSyntaxExtensions.cs

```diff
@@ -115,5 +115,10 @@ public static ObjectSyntax DeepMerge(this ObjectSyntax? sourceObject, ObjectSynt
                     ? mergedObject.MergeProperty(propertyName, property.Value)
                     : mergedObject);
         }
+
+        public static DecoratorSyntax? SafeGetDecoaratorByName(this StatementSyntax syntax, string name)
+        {
+            return syntax.Decorators.FirstOrDefault(d => d.Expression is FunctionCallSyntax func && string.Equals(func.Name.IdentifierName, name, System.StringComparison.OrdinalIgnoreCase));
```

Comment:
This won't work for namespaced decorators - e.g.:
```bicep
@sys.secure()
```

Reply:
Yeah, it's also possible that in the future some other namespace add a `secure` decorator, so we may need to get the decorator function symbol first and search in the `sys` namespace:
```C#
return syntax.Decorators.FirstOrDefault(decoratorSyntax =>
{
    if (this.binder.GetSymbolInfo(decoratorSyntax.Expression) is FunctionSymbol functionSymbol)
    {
        var argumentTypes = this.GetRecoveredArgumentTypes(decoratorSyntax.Arguments).ToArray();
        var decorator = this.binder.FileSymbol.ImportedNamespaces["sys"].Type.DecoratorResolver
            .GetMatches(functionSymbol, argumentTypes)
            .SingleOrDefault();

        if (decorator?.Overload.Name == name)
        {
            return true;
        }
    }

    return false;
});
```

Reply:
Ok, I see the issue. I'd need to move checking decorator/object modifier from ParameterDeclarationSyntax up to TypeAssignmentVisitor and alter the type/flag there if certain decorator/modifier is found.

TBH, similar to getSecret function we should make decorator builder to define (via lambda) how the type is going to change if the decorator it's being used, rather than hard-coding type logic in the TypeAssignmentVisitor.

Reply:
@shenglol - the code you provided does not work.
`this.binder.GetSymbolInfo(decoratorSyntax.Expression)` is returning null.

I as you use same code to check for allowed decorator I checked with the 'stock' version and I get no error when I provide a non-allowed value when using `@sys.allowed()` decorator.